pkg/storage/bloom/v1/builder.go

```diff
@@ -545,8 +549,17 @@ func (mb *MergeBuilder) Build(builder *BlockBuilder) (uint32, error) {
 		nextInBlocks *SeriesWithBloom
 	)
 
+	// reset all the blocks to the start
+	blocksAsPeekIters := make([]PeekingIterator[*SeriesWithBloom], 0, len(mb.blocks))
```

Comment:
Moved here
https://github.com/grafana/loki/blob/1748c29af710c310e24298be8e9398d172f2851d/pkg/bloomcompactor/spec.go#L210-L216

> Do you really need to make a copy of the slice?

Unfortunately yes. We need to pass `[]PeekingIterator` instead of []v1.PeekingResetIter to the builder. I expected Go to infer that they are compatible types but... no :(

Reply:
Hm interesting. Can we cast it?